pkg/util/fileutil/fileutil.go

```
@@ -202,14 +206,118 @@ var _ fs.FileInfo = FsFileInfo{}
+}
+
+func ReadStreamToFileData(ctx context.Context, readCh <-chan wshrpc.RespOrErrorUnion[wshrpc.FileData]) (*wshrpc.FileData, error) {
+	var fileData *wshrpc.FileData
+	var dataBuf bytes.Buffer
+	var entries []*wshrpc.FileInfo
+	err := ReadFileStream(ctx, readCh, func(finfo wshrpc.FileInfo) {
+		fileData = &wshrpc.FileData{
+			Info: &finfo,
+		}
+	}, func(entries []*wshrpc.FileInfo) error {
+		entries = append(entries, entries...)
```

golangci-lint (1.62.2)

ineffectual assignment to entries

(ineffassign)